src/main/java/loci/common/Location.java

```diff
@@ -145,7 +129,40 @@ public Location(File file) {
    * @param child the relative path name
    */
   public Location(String parent, String child) {
-    this(parent + File.separator + child);
+    LOGGER.trace("Location({}, {})", parent, child);
+
+    String mapped = null;
+    String pathname = null;
+
+    // First handle possible URIs
+    if (child.contains("://")) {
```

Comment:
Under some conditions this seem to trigger a NPE - see https://web-proxy.openmicroscopy.org/east-ci/job/BIOFORMATS-test-folder/35481/console

Reply:
Ha. Ok. That's interesting.

https://github.com/snoopycrimecop/bioformats/blob/develop/merge/daily/components/formats-gpl/src/loci/formats/in/InCellReader.java#L863 does

```
      else if (qName.equals("Image")) {
        String file = attributes.getValue("filename");
        String thumb = attributes.getValue("thumbnail");
        Location current = new Location(currentId).getAbsoluteFile();

        Location imageFile = new Location(current.getParentFile(), file);
        currentImageFile = imageFile.getAbsolutePath();
        currentThumbnail =
          new Location(current.getParentFile(), thumb).getAbsolutePath();
      }
```

and https://github.com/ome/ome-common-java/blob/master/src/main/java/loci/common/Location.java#L148 does:

```
    this(parent + File.separator + child);
```

so I assume the current behavior is to return a Location ending in "null". I'll add the test for null here, but this probably could reuse further review.